### DISCUSSION TITLE
The VS 2022 implementation is so shoddy, attempting to charge $10/mo for this is honestly a massive insult..

### DISCUSSION THREAD

**Discussion by MulleDK19** [community/community#31404] [Copilot Conversations]
I wrote this on August 22, the last day of my 'extended access', and I didn't want to update the post to change the tenses from present to past, so keep that in mind.

---
While the VS Code implementation is decent, the VS 2022 implementation is shoddy at best, and borderline useless.

Now, to rule out as much as possible that it's just on my system, I've tried with and without other extensions, reset and reinstalled both Copilot and VS several times, and the issues remain: It just doesn't work well in VS 2022.

- It dies A LOT. I constantly have to re-open the current document, and often completely restart VS before it'll suggest anything.

- It won't suggest anything until you specifically type a space, so you constantly have to type a space and remove it to even get it to suggest anything.

- It very often won't suggest anything on empty lines. Eg. in VS Code, you just hit enter and it'll fill the next line, but in VS, you often have to type something yourself first, eg. `Console.` or `float`, etc. which severely limits what it can suggest for you.

- When rebinding the suggestion key, the new shortcut is ignored.

- The default shortcut rarely seems to work either.

- It seems to be a different engine than VS Code. Two completely identical prompts can return different (and often better) code in VS Code.

- It often suggests my own code that I've typed previously and then deleted, or which is still there, and now getting suggested in invalid positions:

![image](https://user-images.githubusercontent.com/3169809/184947294-2cced028-de7b-4f3a-93f9-1589c7c07f8e.png)

![image](https://user-images.githubusercontent.com/3169809/185729737-27f1ad37-bebc-4977-b431-3ccb8f7cb7db.png)

- If often suggests non-sense

![image](https://user-images.githubusercontent.com/3169809/185734028-96c19e48-fb1c-4143-8c0b-33532be8095e.png)

- It constantly conflicts with IntelliSense and snippets. It won't suggest while IntelliSense is open, and if you type something that's also a snippet, eg. `for` and it then suggests, when you click, the suggestion is removed and the snippet is inserted instead.

- There's absolutely no feedback. You have no idea whatsoever why it won't suggest something, and you're just left wondering if it failed, or you hit the content filter, or if it's just broken. A lot, the Copilot icon is spinning for a long time, and you have no idea whether it's just slow right now, or just not working at all. Which mostly seem to be the latter. If you use "Open Copilot", it either finishes instantly with no suggestions, or it says "Loading Suggestions.." forever, until you restart VS. Often, the icon will do nothing at all, and then when you click it, it's grayed out and you have to restart VS:

![image](https://user-images.githubusercontent.com/3169809/184949987-5c3bc488-e1de-4111-9f0c-ece0aa9238ae.png)

- The content filter is abysmal. This doesn't apply to just VS 2022. The fact that my code can't contain `Gender`, `Male`, `Female`, `Woman`, `Girl`, `Sex`, etc. is just plain stupid.. "Oh, you got a Gender enum? Oh, you're writing a bad word filter? Screw you! No more suggestions for you!"...
No, seriously. I'm working on a Spider-Man tool, and if I maintain the hyphen in my code, Copilot refuses to do shit..

- Today's the last day, and every time I open up a document, Copilot shows a BLOCKING message I have to dismiss every single time, that it expires today..
Oh, and not just when I open a document.. I alt+tabbed for like 10 minutes, and when I switched back to VS, I had to dismiss 3 messages..

I'm constantly finding myself copying and pasting code from VS 2022 into VS Code to perform a suggestion, and then copy it back.
Overall, it's a really poor, useless experience, and trying to charge $10/mo for it is an insult of the highest order..

And 'extended access'? You can barely have called the last few months "access".

I *was* going to record a video showing how much worse the experience is, and how much slower you work with Copilot in VS 2022, but as it turns out, I haven't been able to get Copilot to work at all in VS 2022 the past couple of days, despite multiple restarts, re-signing in, reinstalling, etc....

To top it all off, the model has taken a very huge nose-dive in quality and accuracy since I started using Copilot: https://github.com/community/community/discussions/19655#discussioncomment-3080700
---

Now that it's expired, I'm living without it, as $10/mo (or even $8⅓/mo) is way too much for a product that barely works. (In my case, even with a perfect implementation, $8⅓/mo is steep for me)
And while I miss it for some things, like boiler plate code and P/Invoke signatures, the implementation is so shoddy, I'm way more efficient without it.
Typing out my code manually is far, far faster than constantly having to add and remove spaces, open and close documents, and restarting VS.

Other posters that have noticed the same:

https://github.com/community/community/discussions/29843
https://github.com/community/community/discussions/30977
https://github.com/community/community/discussions/30842

**Comment by Matt-17** [comment]
After reading your thread I tried Copilot with VS Code and now I am blasted. This is a difference like day and night and this annoys me now. I am paying for VS2022 and I get an alpha version? The product is not even more stable, the same file with cursor on the same position gives totally different suggestions (sometimes it's only an open brace in VS 2022 vs. a complete 10 liner in VS Code).

**Reply by Bortronx** [reply]
> After reading your thread I tried Copilot with VS Code and now I am blasted. This is a difference like day and night and this annoys me now. I am paying for VS2022 and I get an alpha version? The product is not even more stable, the same file with cursor on the same position gives totally different suggestions (sometimes it's only an open brace in VS 2022 vs. a complete 10 liner in VS Code).

This is my exact situation. I paid for VS2022 copilot right after reading the blog only to find out that it was false advertising. It doesn't work like the blog describes. To me I have tried updating and doing older version and nothing.

**Reply by tilmann1** [reply]
Can you please update to the latest version (1.43) and try again? Thanks!

**Reply by chenhw** [reply]
> Can you please update to the latest version (1.43) and try again? Thanks!

the issues remain

**Reply by anthonyy41** [reply]
The same problem and often encounter suggestions that cannot be displayed (icon stays in this state)
![image](https://user-images.githubusercontent.com/31025028/188250572-a7cb370d-0f5b-486c-b635-0d99bde391ea.png)

**Reply by tilmann1** [reply]
Please try the just released Copilot 1.44.0.0 VS extension which contains various improvements that help address this issue on VS 17.3.x. Thanks!

**Reply by Matt-17** [reply]
As mentioned in the other thread it seems much better now. Thank you

**Reply by MulleDK19** [reply]
> Please try the just released Copilot 1.44.0.0 VS extension which contains various improvements that help address this issue on VS 17.3.x. Thanks!

That's difficult considering it's a paid service I'm not subscribed to...

**Comment by iamcjwilliams** [comment]
@MulleDK19 @Bortronx @Matt-17 We're sorry for the trouble here. Our Copilot team is working to improve the quality of the extension.

**Reply by Bortronx** [reply]
I understand and I respect that but I paid for this service and that's why I opened a ticket that hasn't been responded to in now over 7 days yet my card was charged from the first day instantly without failure but the feature I was paying for as advertised is not actually available. Unacceptable for a paid service.
